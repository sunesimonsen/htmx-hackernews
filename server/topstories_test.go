package server

import (
	"testing"
)

func TestGetTopStoryIds(t *testing.T) {
	t.Run("with a succesful response", func(t *testing.T) {
		snapshotResponse(t, "/topstories", "/v0/topstories.json", []int{42, 3545, 345, 1})
	})
}
