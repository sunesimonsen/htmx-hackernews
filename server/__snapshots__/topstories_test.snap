
[TestGetTopStoryIds/with_a_succesful_response - 1]
http.Header{
    "Cache-Control": {"max-age=0"},
    "Content-Type":  {"text/html"},
}
int(200)

    
        <article class="story" hx-get="/parts/story/42" hx-trigger="revealed" hx-swap="outerHTML">
            <a class="title"><div class="skeleton" style="width: 60%"></div></a>
            <footer><div class="skeleton" style="width: 40%"></div></footer>
        </article>
    
        <article class="story" hx-get="/parts/story/3545" hx-trigger="revealed" hx-swap="outerHTML">
            <a class="title"><div class="skeleton" style="width: 60%"></div></a>
            <footer><div class="skeleton" style="width: 40%"></div></footer>
        </article>
    
        <article class="story" hx-get="/parts/story/345" hx-trigger="revealed" hx-swap="outerHTML">
            <a class="title"><div class="skeleton" style="width: 60%"></div></a>
            <footer><div class="skeleton" style="width: 40%"></div></footer>
        </article>
    
        <article class="story" hx-get="/parts/story/1" hx-trigger="revealed" hx-swap="outerHTML">
            <a class="title"><div class="skeleton" style="width: 60%"></div></a>
            <footer><div class="skeleton" style="width: 40%"></div></footer>
        </article>
    

---
